src/IO/ImageBase/ProcessImageChunks/Code.cxx

```diff
@@ -0,0 +1,129 @@
+
+int
+main(int argc, char * argv[])
+{
+  if (argc != 3)
+  {
+    std::cerr << "Usage: " << std::endl;
+    std::cerr << argv[0];
+    std::cerr << " <InputFileName> <OutputFileName>";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  const char * inputFileName = argv[1];
+  const char * outputFileName = argv[2];
+
+  const int xDiv = 6;
+  const int yDiv = 4;
+  const int zDiv = 1; // 1 for 2D input
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = unsigned char;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  using ReaderType = itk::ImageFileReader<ImageType>;
+  ReaderType::Pointer reader = ReaderType::New();
+  reader->SetFileName(inputFileName);
+  reader->UpdateOutputInformation();
+  ImageType::RegionType fullRegion = reader->GetOutput()->GetLargestPossibleRegion();
+  ImageType::SizeType   fullSize = fullRegion.GetSize();
+  // when reading image from file, start index is always 0
+
+  ImageType::IndexType start;
+  ImageType::IndexType end;
+  ImageType::SizeType  size;
+
+  using MedianType = itk::MedianImageFilter<ImageType, ImageType>;
+  MedianType::Pointer median = MedianType::New();
+  median->SetInput(reader->GetOutput());
+  median->SetRadius(2);
+
+  using WriterType = itk::ImageFileWriter<ImageType>;
+  WriterType::Pointer writer = WriterType::New();
+  writer->SetFileName(outputFileName);
+
+  // use for loops to split the image into chunks
+  // this way of splitting gives us easy control over it
+  for (int z = 0; z < zDiv; z++)
+  {
+    start[2] = fullSize[2] * double(z) / zDiv;
+    end[2] = fullSize[2] * (z + 1.0) / zDiv;
+    size[2] = end[2] - start[2];
+
+    for (int y = 0; y < yDiv; y++)
+    {
+      start[1] = fullSize[1] * double(y) / yDiv;
+      end[1] = fullSize[1] * (y + 1.0) / yDiv;
+      size[1] = end[1] - start[1];
+
+      for (int x = 0; x < xDiv; x++)
+      {
+        start[0] = fullSize[0] * double(x) / xDiv;
+        end[0] = fullSize[0] * (x + 1.0) / xDiv;
+        size[0] = end[0] - start[0];
+
+        ImageType::RegionType region;
+        region.SetIndex(start);
+        region.SetSize(size);
+
+        // now that we have our chunk, request the filter to only process that
+        median->GetOutput()->SetRequestedRegion(region);
+        median->Update();
+        ImageType::Pointer result = median->GetOutput();
+
+        // only needed in case of further manual processing
+        result->DisconnectPipeline();
+
+        // possible additional non-ITK pipeline processing
+
+        writer->SetInput(result);
+
+        // convert region into IO region
+        itk::ImageIORegion ioRegion(Dimension);
+        ioRegion.SetIndex(0, start[0]);
+        ioRegion.SetIndex(1, start[1]);
+        ioRegion.SetIndex(2, start[2]);
+        ioRegion.SetSize(0, region.GetSize()[0]);
+        ioRegion.SetSize(1, region.GetSize()[1]);
+        ioRegion.SetSize(2, region.GetSize()[2]);
+        // tell the writer this is only a chunk of a larger image
```

Comment:
The ImageRegionSplitterMultidimensional provides vary similar functionality to this manual splitting done above. I see the comment reference "control". Perhaps at build in ITK functionality should at least be referenced even if it's not used.

Reply:
OK, I guess I will also add a more standard ITK streaming pipeline, which has less manual code but would probably be less clear to newcomers.

Reply:
I realized that https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/src/Core/Common/StreamAPipeline/Code.cxx already demonstrates a regular pipeline. Also, this example is meant more for use from Python as part of ML approach. I updated the comments accordingly.